test/acceptance/features/bindAppToServiceAnnotations.feature

```diff
@@ -0,0 +1,434 @@
+        And The application env var "BACKEND_URL_2" has value "black-hole.example.com"
+
+    Scenario: Each value in referred slice of maps from service resource gets injected into app as separate env variable
+        Given Generic test application "slom-app" is running
+        And The Custom Resource Definition is present
+            """
+            apiVersion: apiextensions.k8s.io/v1beta1
+            kind: CustomResourceDefinition
+            metadata:
+                name: backends.stable.example.com
+                annotations:
+                    service.binding/webarrows: path={.spec.connections},elementType=sliceOfMaps,sourceKey=type,sourceValue=url
+            spec:
+                group: stable.example.com
+                versions:
+                  - name: v1
+                    served: true
+                    storage: true
+                scope: Namespaced
+                names:
+                    plural: backends
+                    singular: backend
+                    kind: Backend
+                    shortNames:
+                      - bk
+            """
+        And The Custom Resource is present
+            """
+            apiVersion: stable.example.com/v1
+            kind: Backend
+            metadata:
+                name: slom-service
+            spec:
+                connections:
+                  - type: primary
+                    url: primary.example.com
+                  - type: secondary
+                    url: secondary.example.com
+                  - type: '404'
+                    url: black-hole.example.com
+            status:
+                somestatus: good
+            """
+        When Service Binding is applied
+            """
+            apiVersion: operators.coreos.com/v1alpha1
+            kind: ServiceBinding
+            metadata:
+                name: slom-binding
+            spec:
+                services:
+                  - group: stable.example.com
+                    version: v1
+                    kind: Backend
+                    name: slom-service
+                application:
+                    name: slom-app
+                    group: apps
+                    version: v1
+                    resource: deployments
+            """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "slom-binding" should be changed to "True"
+        And jq ".status.conditions[] | select(.type=="InjectionReady").status" of Service Binding "slom-binding" should be changed to "True"
+        And The application env var "BACKEND_WEBARROWS_PRIMARY" has value "primary.example.com"
+        And The application env var "BACKEND_WEBARROWS_SECONDARY" has value "secondary.example.com"
+        And The application env var "BACKEND_WEBARROWS_404" has value "black-hole.example.com"
+
+    Scenario: Backend Service metadata annotations update for service bindings gets propagated to the binding secret
```

Comment:
This test adds another annotation `service.binding/ready: path={.status.ready}` rather than updating an existing annotation i.e. `service.binding/host: path={.spec.host}`.

So, list of annotations are *updated* definitely.. but we're not showing update to a single annotation.

Reply:
@DhritiShikhar This scenario was added by @Avni-Sharma as part of https://github.com/redhat-developer/service-binding-operator/pull/704. In this PR (beyond adding the new CRD annotation scenarios) I've just moved the other annotation-related scenarios into a separate file. So a fix to this (if needed) should go in a separate PR (as a follow-up), to keep PRs small. Would you agree?

Reply:
Agreed. I will create a github issue for that.

Reply:
Github issue for the same -> https://github.com/redhat-developer/service-binding-operator/issues/734

Reply:
Thx